File: Core/src/main/java/dev/lynith/core/hud/InGameHud.java
package dev.lynith.core.hud;

import dev.lynith.core.ui.Screen;
import dev.lynith.core.ui.styles.ComponentStyles;
import dev.lynith.core.utils.GuiScreens;
import dev.lynith.core.utils.ZabuColor;
import dev.lynith.core.versions.renderer.IRenderer;

public class InGameHud extends Screen<InGameHud, ComponentStyles<InGameHud>> {

    @Override
    public void render(IRenderer ctx) {
        if (ctx.getCurrentScreenType() != GuiScreens.HUD_CONFIG_SCREEN) {
            super.render(ctx);
        }
    }

    @Override
    public void init() {
        add(HudManager.getInstance().getComponents());

        style(styles -> {
            styles.background(ZabuColor.TRANSPARENT);
        });

        super.init();
    }
}

File: Core/src/main/kotlin/dev/lynith/core/hud/InGameHUD.kt
package dev.lynith.core.hud

import dev.lynith.core.Platform
import dev.lynith.core.bridge.gui.IRenderer
import dev.lynith.core.bridge.gui.MCScreen
import dev.lynith.core.ui.components.Screen
import dev.lynith.core.ui.styles.impl.Color

class InGameHUD : Screen() {

    override fun render(mouseX: Int, mouseY: Int, delta: Float) {
        if (Screen.currentScreen?.javaClass != HudConfigScreen::class.java) {
            Platform.hudManager.render(delta)
        }
    }

    override fun init() {

    }

    override fun toMCScreen(): MCScreen {
        return WrappedScreen(this)
    }

    private class WrappedScreen(val screen: Screen) : Screen.WrappedScreen(screen) {
        override fun render(mouseX: Int, mouseY: Int, delta: Float) {
            screen.wrappedRender(mouseX, mouseY, delta)
        }
    }

}